src/components/Text.js

```diff
@@ -35,7 +37,16 @@ const Text = ({
       hyphenation && 'hyphenation'
     )}
   >
-    {children}
+    {orphans ? (
+      children
+    ) : (
+      <span
+        dangerouslySetInnerHTML={{
+          __html: children.toString().replace(/ ([^ ]*)$/, '&nbsp;$1'),
```

Comment:
This feels alarming, this opens us up to some xss, of which we weren't _as_ vulnerable as before.
I have made contrived example of why this is dangerous: https://codesandbox.io/s/misty-paper-e2wk0

Essentially, if we merge this we have to agree that we 100% trust that the text component will:
- _only_ receive a string of text
- the string will **not** contain malicious HTML

I don't think the trade off is worth it, i.e. this bit of security in exchange for no orphan words.

Reply:
> only receive a string of text

In the case that `orphans={true}`. We could/should add additional type validation to ensure that when this is the case, we also check that children is of type string. As it stands it's not a great API though to be honest.

> the string will not contain malicious HTML

My understanding was that we could always trust our content here. In that it should always come from the CMS and we trust each other not to enter something malicious? If that's not the case though then yeah we should obviously not do this :slightly_smiling_face:

Also because it's awkward and complicates the API, sadly, I would also prefer not to have to support this requirement. With some time I/we might be able to investigate a better solution though.

Reply:
I've created an example which doesn't use `dangerouslySetInnerHTML` https://codesandbox.io/s/clever-hooks-fbrcq. Take a look at the NotDanger component. Would this work for us here?